src/utility/api.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * Question: why does this file exist?
+ * Answer: The fetch API, while great, has limitations. In particular, it does not throw a rejected promise when
+ * a non 200 response is received and does not automatically unmarshal JSON. This wrapper solves those problems and
+ * provides a uniform API, returning an object of the form { json, error, response } in all cases, whether the promise
+ * was rejected or not. In some cases, we still want access to the json even if the response is not 200.
+ * 
+ * This gives you the original response as well, in case you want that.
+ * 
+ * This wrapper is also designed to interface nicely with the URL API implemented in url.js.
+ * 
+ * This wrapper will *not* work as intended for response types that are not JSON or empty.
+ */
+
+const REQUEST_VERBS = Object.freeze({
```

Comment:
if we do a `constants.js` (suggested below) perhaps put this (and the ones right below) in there

Reply:
I think I'll keep these here, since I see them as (private) implementation details of the API.